lib/providers/gcm.js

```diff
@@ -70,6 +70,7 @@ GcmProvider.prototype._createMessage = function(notification) {
   Object.keys(notification).forEach(function (key) {
     if (notification[key] !== null && typeof notification[key] !== 'undefined') {
       message.addData(key, notification[key]);
+      message.addNotification(key, notification[key]);
```

Comment:
The only thing you should be aware of is that the notification must contain
body and title properties

On Mon, Aug 3, 2015 at 6:20 PM Raymond Feng <notifications@github.com>
wrote:

> In lib/providers/gcm.js
> <https://github.com/strongloop/loopback-component-push/pull/98#discussion_r36095782>
> :
>
> > @@ -70,6 +70,7 @@ GcmProvider.prototype._createMessage = function(notification) {
> >    Object.keys(notification).forEach(function (key) {
> >      if (notification[key] !== null && typeof notification[key] !== 'undefined') {
> >        message.addData(key, notification[key]);
> > +      message.addNotification(key, notification[key]);
>
> I was wondering if we should allow the notification argument to have
> separate data and notification properties, for example:
>
> {
>   x: 1,
>   notifications: {
>     y: 2
>   }
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/loopback-component-push/pull/98/files#r36095782>
> .
>